Buildings/BoundaryConditions/SolarGeometry/ProjectedShadowLength.mo

```diff
@@ -68,15 +68,11 @@ protected
   BaseClasses.SolarHourAngle solHouAng "Solar hour angle"
     annotation (Placement(transformation(extent={{60,110},{80,130}})));
 
-  Modelica.Blocks.Logical.LessThreshold lessThreshold(
-    final threshold=0.5*Modelica.Constants.pi)
-    "Output zero if sun is below horizon"
-    annotation (Placement(transformation(extent={{-30,-120},{-10,-100}})));
-  Modelica.Blocks.Logical.Switch switch1 "Switch to select the output signal"
-    annotation (Placement(transformation(extent={{120,-120},{140,-100}})));
-
-  Modelica.Blocks.Sources.Constant zer(final k=0) "Zero output"
-    annotation (Placement(transformation(extent={{60,-140},{80,-120}})));
+public
```

Comment:
Please make this `RealExpression` protected. I don't see what we gain by making it public.

Also, make the line break somewhere where it is easier to read, such as after `pShaLen`

And it should be `pShaLen`, not `pshaLen`, no?